rimage/transform/parallel_projection.go

```diff
@@ -95,5 +97,162 @@ func (pp *ParallelProjection) PointCloudToRGBD(cloud pointcloud.PointCloud) (*ri
 
 // ImagePointTo3DPoint takes the 2D pixel point and assumes that it represents the X,Y coordinate in mm as well.
 func (pp *ParallelProjection) ImagePointTo3DPoint(pt image.Point, d rimage.Depth) (r3.Vector, error) {
-	return r3.Vector{float64(pt.X), float64(pt.Y), float64(d)}, nil
+	return r3.Vector{X: float64(pt.X), Y: float64(pt.Y), Z: float64(d)}, nil
+}
+
+// ParallelProjectionOntoXZWithRobotMarker allows the creation of a 2D projection of a pointcloud and robot
+// position onto the XZ plane.
+type ParallelProjectionOntoXZWithRobotMarker struct {
+	robotPose *spatialmath.Pose
+}
+
+const (
+	sigmaLevel       = 7
+	imageHeight      = 1080
+	imageWidth       = 1070
+	missThreshold    = 0.44
+	hitThreshold     = 0.53
+	voxelRadius      = 1
+	robotMarkerRatio = 5
+)
+
+// PointCloudToRGBD creates an image of a pointcloud in the XZ plane, scaling the points to a standard image
+// size. It will also add a red marker to the map to represent the location of the robot. The returned depthMap
+// is unused and so will always be nil.
+func (ppRM *ParallelProjectionOntoXZWithRobotMarker) PointCloudToRGBD(cloud pointcloud.PointCloud,
+) (*rimage.Image, *rimage.DepthMap, error) {
```

Comment:
Is there a reason why we are returning pointers to the images & depthmap as opposed to the values themselves? Is this for performance reasons, a go idiom I haven't learned yet, or another reason?

Reply:
Good question! It is due to the large size of images typically making it more efficient to pass pointers to it around as opposed to the image itself.

In any event, we need to to abide by the interface and chaning the interface is out of scope for this project